### DISCUSSION TITLE
IOS and unity

### DISCUSSION THREAD

**Discussion by adenek** [Tyrrrz/YoutubeExplode#693] [Q&A]
Hello,

I'm using YoutubeExplode in unity for android and it work like a charm. 
I'm now trying to port my game to IOS but I do have this error even in the MaxOS unity editor

ObjectDisposedException: Cannot access a closed Stream.
System.IO.MemoryStream.EnsureNotClosed () (at <5b5cf4db41e44e579e17cb24a65f9a93>:0)
System.IO.MemoryStream.Read (System.Byte[] buffer, System.Int32 offset, System.Int32 count) (at <5b5cf4db41e44e579e17cb24a65f9a93>:0)
System.IO.MemoryStream.ReadAsync (System.Byte[] buffer, System.Int32 offset, System.Int32 count, System.Threading.CancellationToken cancellationToken) (at <5b5cf4db41e44e579e17cb24a65f9a93>:0)

This error is related to:
YoutubeExplode.Videos.Streams.MediaStream.ReadSegmentAsync (System.Byte[] buffer, System.Int32 offset, System.Int32 count, System.Threading.CancellationToken cancellationToken) (at D:/a/YoutubeExplode/YoutubeExplode/YoutubeExplode/Videos/Streams/MediaStream.cs:84)
YoutubeExplode.Videos.Streams.MediaStream.ReadAsync (System.Byte[] buffer, System.Int32 offset, System.Int32 count, System.Threading.CancellationToken cancellationToken) (at 

Do you guys have any idea of how I could fix that?

Thx

**Comment by steve3p0** [comment]
Did you ever get this resolved?
